go/types/encode_human_readable.go

```diff
@@ -246,7 +246,10 @@ func (w *hrsWriter) writeType(t *Type, parentStructTypes []*Type) {
 		w.write("<")
 		for i, et := range t.Desc.(CompoundDesc).ElemTypes {
 			if i != 0 {
-				w.write(", ")
+				w.write(",")
+				if et.Kind() != UnionKind {
```

Comment:
I don't think this will do the right thing in case of non empty unions. Desired output:

```
Map<T, V | W>
Map<T,>
```

Maybe add a test for this too?